force-app/main/default/classes/LeadTriggerHandlerTest.cls

```diff
@@ -0,0 +1,114 @@
+@isTest
+private class LeadTriggerHandlerTest {
+
+    @isTest
+    static void testHandleTitleNormalization() {
+        // Create test data (Leads)
+        List<Lead> testLeads = new List<Lead>();
+        // Add your test data here (e.g., create Leads with different titles)
+        Lead lead1 = new Lead(Title = 'VP of Sales', LastName = 'Test1', Company = 'Test Company1');
+        Lead lead2 = new Lead(Title = 'Sales Manager', LastName = 'Test2', Company = 'Test Company2');
+        Lead lead5 = new Lead(Title = 'Business Owner', LastName = 'Test2', Company = 'Test Company5');
+        Lead lead3 = new Lead(Title = 'Marketing', LastName = 'Test3' , Company = 'Test Company3');
+        Lead lead4 = new Lead(Title = 'Manager - Marketing', LastName = 'Test4', Company = 'Test Company4');
+        Lead lead6 = new Lead(Title = 'head', LastName = 'Test6', Company = 'Test Company6');
+        Lead lead7 = new Lead(Title = 'jr', LastName = 'Test7', Company = 'Test Company7');
+        
+        testLeads.add(lead1);
+        testLeads.add(lead2);
+        testLeads.add(lead3);
+        testLeads.add(lead4);
+        testLeads.add(lead5);
+        testLeads.add(lead6);
+        testLeads.add(lead7);
+
+        insert testLeads;
+
+        // Call the trigger handler method
+        LeadTriggerHandler handler = new LeadTriggerHandler(true, 200);
+        handler.handleTitleNormalization(testLeads);
```

Comment:
This works but it's doing doubling the handleTitleNormalization call because you are inserting the leads then calling the handlers. 
Just try calling the noninserted leads with the handler